### ISSUE TITLE
More modularization for metrics and logs

### ISSUE THREAD

**Issue by anvari1313** [snapp-incubator/qsse#50]
It's a good idea to make some functionalities like Prometheus metrics and logging modular and enable the users to customize them in their own way.

**Comment by 1995parham** [comment]
Yes, exactly right now Golang doesn't have any standard logger interface that support structured logging, so we force using zap as logger but for the Prometheus part I think open-telemetry metrics interface is great solution. If all agree, we can use that.

**Comment by anvari1313** [comment]
Or we can think about implementing an interceptor-like pattern in the project to make it extendable
